autoPyTorch/api/base_task.py

```diff
@@ -1,3 +1,18 @@
+"""Base class for tasks to solve
+* The shared components among all the tasks
+* This module provides the optimization given a pipeline
```

Comment:
I think this line is a bit confusing. We can mention that it initiates HPO for the pipeline search space.